scripts/dockerbuild.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+set -e
+
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ "$SOURCE" != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+
+cd "$DIR/.."
+
+INTERACTIVE="-i"
+
+while [[ $# > 0 ]]; do
+    key=$1
+
+    case $key in
+        --non-interactive)
+            INTERACTIVE=
+            ;;
+        -t|--tag)
+            DOCKER_TAG=$2
+            shift
+            ;;
+        -d|--dockerfile)
+            DOCKERFILE=$2
+            shift
+            ;;
+        -h|-?|--help)
+            echo "Usage: $0 [-d|--dockerfile <Dockerfile>] [-t|--tag <Tag>] <Command>"
+            echo ""
+            echo "Options:"
+            echo "  <Dockerfile>    The path to the folder that contains a Dockerfile to use to create the build container"
+            echo "  <Tag>           The name of docker image tag"
+            exit 0
+            ;;
+        *)
+            break # the first non-switch we get ends parsing
+            ;;
+    esac
+
+    shift
+done
+
+# Make container names CI-specific if we're running in CI
+#  Jenkins
+[ ! -z "$BUILD_TAG" ] && DOTNET_BUILD_CONTAINER_NAME="${BUILD_TAG}-${DOCKER_TAG}"
+#  VSO
+[ ! -z "$BUILD_BUILDID" ] && DOTNET_BUILD_CONTAINER_NAME="${BUILD_BUILDID}-${BUILD_BUILDNUMBER}-${DOCKER_TAG}"
+
+# Executes a command and retries if it fails.
+# NOTE: This function is the exact copy from init-docker.sh.
+# Reason for not invoking init.docker.sh directly is since that script 
+# also performs cleanup, which we do not want in this case.
+execute() {
```

Comment:
While I agree 1719 is the best long term fix, the reality is it will probably be some time before that happens.  Duplicating this script logic is problematic in the interim.  Refactoring it now looks pretty easy.